### PR TITLE
Add Ruby VM specific metrics

### PR DESCRIPTION
**Pull request by mikker**
Continued from elastic/apm-agent-ruby#437 because Jenkins

### CODE REVIEW

File: spec/elastic_apm_spec.rb [hunk 1]
```diff
@@ -14,7 +14,10 @@
   end
 
   context 'when running', :mock_intake do
-    before { ElasticAPM.start }
+    before do
+      # disable immediate metrics collection
+      ElasticAPM.start(metrics_interval: 0)
```
Comment (mikker) [inline spec/elastic_apm_spec.rb]: @estolfo Not really a fan but it solves the JRuby test failures. The delegation tests are already a bit weird (although they have discovered errors before.)

_Wdyt?_

**Comment by mikker** [comment]
@estolfo Merging this as it was already reviewed and approved from @axw in its original PR.

The last many commits were just fighting with either JRuby or Jenkins which was quite a bit frustrating. Perhaps we can reevaluate this fix later for example https://github.com/elastic/apm-agent-ruby/pull/487/files#diff-6ad466b0c60404c7ae066e0168d3e910R53-R61

Testing it in a production app over the weekend and if everything looks good, doing a minor release on Monday.

File: lib/elastic_apm/metrics/vm.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module ElasticAPM
+  module Metrics
+    # @api private
+    class VM
+      include Logging
+
+      def initialize(config)
```
Comment (estolfo) [inline lib/elastic_apm/metrics/vm.rb]: The `config` argument is not captured. Is that intentional? `config` is an `attr_reader` so I'm assuming it should be captured.

Comment (mikker) [reply lib/elastic_apm/metrics/vm.rb]: IIRC, the `Logging` module expects a `config` method to be available. It's something that I eventually want to rewrite.

Comment (mikker) [reply lib/elastic_apm/metrics/vm.rb]: ... but it should of course be saved to an ivar, yes, good catch!

Comment (mikker) [reply lib/elastic_apm/metrics/vm.rb]: https://github.com/elastic/apm-agent-ruby/pull/519

**Pull request merged**